### PR TITLE
Make MOSN support Istio1.4

### PR DESCRIPTION
**Pull request by peacocktrain**
### Solutions

a new pr for merge

### CODE REVIEW

**Comment by wangfakang** [comment]
These are related to PR review records https://github.com/mosn/mosn/pull/1025 and https://github.com/mosn/mosn/pull/1001.

File: pkg/xds/conv/convertxds.go [hunk 1]
```diff
@@ -1040,21 +1051,18 @@ func convertHealthChecks(xdsHealthChecks []*xdscore.HealthCheck) v2.HealthCheck
...
 }
 
 func convertCircuitBreakers(xdsCircuitBreaker *xdscluster.CircuitBreakers) v2.CircuitBreakers {
-	if xdsCircuitBreaker == nil || xdsCircuitBreaker.Size() == 0 {
+	if xdsCircuitBreaker == nil {
 		return v2.CircuitBreakers{}
 	}
 	thresholds := make([]v2.Thresholds, 0, len(xdsCircuitBreaker.GetThresholds()))
 	for _, xdsThreshold := range xdsCircuitBreaker.GetThresholds() {
-		if xdsThreshold.Size() == 0 {
```
[the remark this answers is not in the document]
Comment (peacocktrain) [reply pkg/xds/conv/convertxds.go]: useless

File: pkg/xds/conv/convertxds.go [hunk 2]
```diff
@@ -1040,21 +1051,18 @@ func convertHealthChecks(xdsHealthChecks []*xdscore.HealthCheck) v2.HealthCheck
...
-		Interval:       *xdsHealthChecks[0].GetInterval(),
-		IntervalJitter: convertDuration(xdsHealthChecks[0].GetIntervalJitter()),
+		Timeout:        ConvertDuration(xdsHealthChecks[0].GetTimeout()),
+		Interval:       ConvertDuration(xdsHealthChecks[0].GetInterval()),
+		IntervalJitter: ConvertDuration(xdsHealthChecks[0].GetIntervalJitter()),
 	}
 }
 
 func convertCircuitBreakers(xdsCircuitBreaker *xdscluster.CircuitBreakers) v2.CircuitBreakers {
-	if xdsCircuitBreaker == nil || xdsCircuitBreaker.Size() == 0 {
```
[the remark this answers is not in the document]
Comment (peacocktrain) [reply pkg/xds/conv/convertxds.go]: proto.Size(xdsCircuitBreaker) fixed

File: pkg/xds/conv/convertxds.go [hunk 3]
```diff
@@ -465,46 +485,65 @@ func convertMixerConfig(s *types.Struct) (map[string]interface{}, error) {
...
-		filterChains = append(filterChains, filterChain)
 	}
-	return filterChains
+
+	return []v2.FilterChain{{
+		FilterChainConfig: v2.FilterChainConfig{
+			FilterChainMatch: chainMatch,
+			Filters:          convertFilters(xdsFilters),
+		},
+		TLSContexts: nil,
```
[the remark this answers is not in the document]
Comment (peacocktrain) [reply pkg/xds/conv/convertxds.go]: the sds，包括tls，Including TLS, requiring a separate module reprogramming, tlscontext has been deprecated

File: pkg/shm/shm_test.go [hunk 4]
```diff
@@ -60,9 +60,9 @@ func TestAtomic(t *testing.T) {
 
 	wg.Wait()
 
-	if *counter != uint32(expected) {
-		t.Errorf("counter error, expected %d, actual %d", 10000, *counter)
-	}
```
[the remark this answers is not in the document]
Comment (peacocktrain) [reply pkg/shm/shm_test.go]: this test failed in my local and now rollback

**Pull request merged**